File: lw6/Garanin_Timofey/DictionaryComressor/DictionaryComressor/RepositoryProcessedData.cpp
#include "stdafx.h"
#include "RepositoryProcessedData.h"

CRepositoryProcessedData::CRepositoryProcessedData()
{
	m_dictionary = std::make_shared<std::unordered_map<std::string, std::string>>();
	m_text = std::make_shared<std::string>();
}

void CRepositoryProcessedData::SetNewDictionaryFragment(std::shared_ptr<std::unordered_map<std::string, std::string>> dictionaryFragment)
{
	m_dictionary = dictionaryFragment;
}

void CRepositoryProcessedData::SetNewTextFragment(std::shared_ptr<std::string> text)
{
	m_text->insert(m_text->end(), text->begin(), text->end());
}

std::shared_ptr<std::unordered_map<std::string, std::string>> CRepositoryProcessedData::GetDictionary() const
{
	return m_dictionary;
}

std::shared_ptr<std::string> CRepositoryProcessedData::GetText() const
{
	return m_text;
}

File: lw7/Garanin Timofey/DictionaryCompressor/DictionaryCompressor/RepositoryProcessedData.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DictionaryCompressor
{
    
    class RepositoryProcessedData
    {
        private Dictionary<string, string> m_dictionary;
        private string m_text;

        public RepositoryProcessedData()
        {
            m_dictionary = new Dictionary<string, string>();
        }

        public void SetNewDictionaryFragment(Dictionary<string, string> dictionaryFragment)
        {
            m_dictionary = dictionaryFragment;
        }

        public void SetNewTextFragment(string text)
        {
            m_text += text;
        }

        public Dictionary<string, string> GetDictionary()
        {
	        return m_dictionary;
        }

        public string GetText()
        {
	        return m_text;
        }
    }
}
